### PR TITLE
[fix lint warnings: NFTs pallet] fix clippy::missing_errors_doc lint warnings

### PR DESCRIPTION
**Pull request by sacha-l**
This PR fixes the lint warnings emitted by the missing_errors_doc lint in the NFTs pallet by adding an # Errors section to the common helper functions.

### CODE REVIEW

**Review by brunopgalvao — APPROVED** [review]
Polishing up the docs on the NFTs pallet nice!

**Review by jsidorenko — APPROVED** [review]

File: frame/nfts/src/features/create_delete_collection.rs [hunk 1]
```diff
@@ -56,6 +66,23 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
...
+	/// This function destroys the specified collection if it exists and meets the necessary
+	/// conditions. It checks the provided `witness` against the actual collection details and
+	/// removes the collection along with its associated metadata, attributes, and configurations.
+	/// The necessary deposits are returned to the corresponding accounts, and the roles and
+	/// configurations for the collection are cleared. Finally, it emits the `Destroyed` event upon
+	/// successful destruction.
+	///
+	/// # Errors
+	///
+	/// This function returns an error in the following cases:
```
Comment (sacha-l) [inline frame/nfts/src/features/create_delete_collection.rs:81]:
```suggestion
	/// This function returns a `Dispatch` error in the following cases:
```

File: frame/nfts/src/features/create_delete_item.rs [hunk 2]
```diff
@@ -19,6 +19,23 @@ use crate::*;
...
+	/// details.
+	///
+	/// This function performs the minting of a new unique item. It checks if the item does not
+	/// already exist in the given collection, and if the max supply limit (if configured) is not
+	/// reached. It also reserves the required deposit for the item and sets the item details
+	/// accordingly.
+	///
+	/// # Errors
+	///
+	/// This function returns an error in the following cases:
```
Comment (sacha-l) [inline frame/nfts/src/features/create_delete_item.rs:32]:
```suggestion
	/// This function returns a `Dispatch` error in the following cases:
```

File: frame/nfts/src/features/create_delete_item.rs [hunk 3]
```diff
@@ -163,6 +180,13 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 		Ok(())
 	}
 
+	/// Burns the specified item with the given `collection`, `item`, and `with_details`.
+	///
+	/// # Errors
+	///
+	/// This function returns an error in the following cases:
```
Comment (sacha-l) [inline frame/nfts/src/features/create_delete_item.rs:187]:
```suggestion
	/// This function returns a `Dispatch` error in the following cases:
```

File: frame/nfts/src/features/transfer.rs [hunk 4]
```diff
@@ -19,6 +19,15 @@ use crate::*;
 use frame_support::pallet_prelude::*;
 
 impl<T: Config<I>, I: 'static> Pallet<T, I> {
+	/// Performs the transfer action for the given `collection`, `item`, `dest`, and `event`.
+	///
+	/// # Errors
+	///
+	/// This function returns an error in the following cases:
```
Comment (sacha-l) [inline frame/nfts/src/features/transfer.rs:26]:
```suggestion
	/// This function returns a `Dispatch` error in the following cases:
```

**Review by alexbcasas — APPROVED** [review]

File: frame/nfts/src/features/attributes.rs [hunk 5]
```diff
@@ -381,21 +381,36 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 		Ok(result)
 	}
 
-	/// A helper method to construct attribute's key.
+	/// A helper method to construct an attribute's key.
+	///
+	/// # Errors
+	///
+	/// This function returns an [`IncorrectData`](crate::Error::IncorrectData) error if the
+	/// provided attribute `key` is incorrectly formatted.
```
Comment (ggwpez) [inline frame/nfts/src/features/attributes.rs:389]:
```suggestion
	/// provided attribute `key` is too long.
```

File: frame/nfts/src/features/attributes.rs [hunk 6]
```diff
@@ -381,21 +381,36 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
...
 	) -> Result<BoundedVec<u8, T::ValueLimit>, DispatchError> {
 		Ok(BoundedVec::try_from(value).map_err(|_| Error::<T, I>::IncorrectData)?)
 	}
 
 	/// A helper method to check whether a system attribute is set for a given item.
+	///
+	/// # Errors
+	///
+	/// This function returns an [`IncorrectData`](crate::Error::IncorrectData) error if the
+	/// provided pallet attribute is incorrectly formatted.
```
Comment (ggwpez) [inline frame/nfts/src/features/attributes.rs:413]:
```suggestion
	/// provided pallet attribute is too long.
```

File: frame/nfts/src/features/metadata.rs [hunk 7]
```diff
@@ -209,6 +209,11 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 	}
 
 	/// A helper method to construct metadata.
+	///
+	/// # Errors
+	///
+	/// This function returns an [`IncorrectMetadata`](crate::Error::IncorrectMetadata) dispatch
+	/// error if the provided metadata is incorrectly formatted.
```
Comment (ggwpez) [inline frame/nfts/src/features/metadata.rs:216]:
```suggestion
	/// error if the provided metadata is too long.
```

**Review by ggwpez — APPROVED** [review]

**Comment by sacha-l** [comment]
bot merge

**Pull request merged**